test-data/unit/check-expressions.test

```diff
@@ -2024,7 +2024,23 @@ cb: Union[Container[A], Container[B]]
 [builtins fixtures/bool.pyi]
 [typing fixtures/typing-full.pyi]
 
-[case testStrictEqualityNoPromote]
+[case testStrictEqualityBytesSpecial]
+# flags: --strict-equality
+b'abc' in b'abcde'
+[builtins fixtures/primitives.pyi]
+[typing fixtures/typing-full.pyi]
+
+[case testStrictEqualityBytesSpecialUnion]
+# flags: --strict-equality
+from typing import Union
+x: Union[bytes, str]
+
+b'abc' in x
```

Comment:
Yes, but this is independent of this flag. The error message says "Non-overlapping container check ..." while in this example the check _may_ return True. I think this can be tightened in typeshed, we can just define `str.__contains__` as accepting `str`, because `42 in 'b'` etc. all fail as well at runtime.